biogrid/libs/biogrid-simulator/src/lib/biogrid/biogrid-class.ts

```diff
@@ -122,23 +153,32 @@ export class Biogrid implements Grid {
    */
   takeAction(action: GridAction) {
     // RETURN a new BiogridState
-    const allSupplyingPaths = action.getSupplyingPaths()
+    const allSupplyingPaths = action.getSupplyingPaths();
 
     const clonedGraph = this.state.cloneStateGraph();
 
     for (const supplyPath in allSupplyingPaths) {
       const oldGridItem = this.state.getGridItem(supplyPath);
```

Comment:
also not part of your PR, but what do you mean by "oldGridItem" here?

Reply:
@RolandNaijuka, also correct me if I am wrong, I believe it has to do with the fact that the grid item gets updated?

Reply:
that is right, this gridItem is updated at the end of the for loop in the clonedGraph